layer/logged_data/queuing_network_log_destination.py

```diff
@@ -0,0 +1,78 @@
+import queue
+import threading
+from queue import Queue
+from typing import Any, Callable, List, Optional
+
+from layer.clients.logged_data_service import LoggedDataClient
+from layer.logged_data.data_logging_request import DataLoggingRequest
+from layer.logged_data.log_data_destination import LogDataDestination
+from layer.logged_data.retrying_files_sender import RetryingFilesSender
+
+
+WAIT_INTERVAL_SECONDS = 1
+
+
+class QueueingNetworkLogDestination(LogDataDestination):
+    def __init__(self, client: LoggedDataClient) -> None:
+        self._log_data_client = client
+        self._files_storage = RetryingFilesSender()
+
+        self._sending_errors: List[Any] = []
+        self._stop_reading = False
+        self._local_queue: Queue[DataLoggingRequest] = Queue()
+        self._reading_thread = threading.Thread(target=self._send_elem_from_queue)
```

Comment:
I don't know the answer to this, but we sometimes use `ThreadPoolExecutor` and sometimes use `Thread`. Just raising this, maybe @donatasm would chime in.

Reply:
If single thread does the job, use `Thread`.

We use `ThreadPoolExecutor`, because it is easier to work with coroutines.

However, while `Thread` is simpler, `ThreadPoolExecutor` would have these advantages:
* increasing number of parallelism is easier
* if thread dies, executor will take care creating another
* executor is easy to change to `ProcessPoolExecutor` if we decide we need move out of proc for queue processing.

Reply:
thank you, I sticked to `Thread`. I think it's good enough for now, and the logic is contained in one class so easy to extend/change in the future if needed.